app/resources/v1/pacbio/run_resource.rb

```diff
@@ -56,6 +56,12 @@ def self.resource_klass_for(type)
 
       def publish_messages
         Messages.publish(@model, Pipelines.pacbio.message)
+        all_library_aliquots = @model.plates.flat_map do |plate|
```

Comment:
all_library_aliquots seems like a method that should be on the model, that way it is easier to test as well.